### PR TITLE
Use colors instead of gulp-util for colorization

### PR DESCRIPTION
**Pull request by nickpape-msft**
* We were having an issue where colors were not being loaded since the gulp binary loads gulp-util which ultimately does the colorization check before our fix in `gulp-core-build` runs.
* After some research, I found that `gulp-util` is deprecated, so this PR replaces all instances where we use it for colorization with the `colors` library used elsewhere. This fixed the above problem.

### CODE REVIEW

File: core-build/gulp-core-build/src/tasks/GulpTask.ts [hunk 1]
```diff
@@ -9,7 +10,8 @@ import { GulpProxy } from '../GulpProxy';
 import { IExecutable } from '../IExecutable';
 import { IBuildConfig } from '../IBuildConfig';
 import { log, verbose, error, fileError, fileWarning,
-  warn, logEndSubtask, logStartSubtask } from '../logging';
+  warn, logEndSubtask, logStartSubtask
+} from '../logging';
```
Comment (iclanton) [inline core-build/gulp-core-build/src/tasks/GulpTask.ts]: The convention is to put these on separate lines.

File: core-build/gulp-core-build/src/logging.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
-import * as gutil from 'gulp-util';
+import * as colors from 'colors';
+import * as fancyLog from 'fancy-log';
 import * as Gulp from 'gulp';
```
Comment (iclanton) [inline core-build/gulp-core-build/src/logging.ts]: Would the guidance be to start removing gulp-util from everywhere?

**Comment by iclanton** [comment]
>     "gulp-util": "~3.0.7",

If this package is deprecated, should we remove it?

---
Refers to: core-build/gulp-core-build/package.json:43 in f4412b5. [](commit_id = f4412b5795766785055932963e10a86ed8bcde40, deletion_comment = False)

**Comment by iclanton** [comment]
>     "node-forge": "~0.7.1",

Same with the other projects.

---
Refers to: core-build/gulp-core-build-serve/package.json:25 in f4412b5. [](commit_id = f4412b5795766785055932963e10a86ed8bcde40, deletion_comment = False)

File: core-build/gulp-core-build-webpack/src/WebpackTask.ts [hunk 3]
```diff
@@ -175,9 +175,9 @@ export class WebpackTask<TExtendedConfig = {}> extends GulpTask<IWebpackTaskConf
                 child.chunks.forEach(chunk => {
                   if (chunk.files && this.taskConfig.printStats) {
                     chunk.files.forEach(file => (
-                      this.log(`Bundled: '${gutil.colors.cyan(path.basename(file))}', ` +
-                                `size: ${gutil.colors.magenta(chunk.size)} bytes, ` +
-                                `took ${gutil.colors.magenta(duration)} ms.`)
+                      this.log(`Bundled: '${colors.cyan(path.basename(file))}', ` +
+                                `size: ${colors.magenta(chunk.size)} bytes, ` +
+                                `took ${colors.magenta(duration.toString(10))} ms.`)
```
Comment (iclanton) [inline core-build/gulp-core-build-webpack/src/WebpackTask.ts]: >duration.toString(10) [](start = 55, length = 21)

Unrelated?

Comment (nickpape-msft) [reply core-build/gulp-core-build-webpack/src/WebpackTask.ts]: It is related. We can't pass a number into colors.

File: core-build/gulp-core-build/src/logging.ts (hunk 2 above)
Comment (nickpape-msft) [reply core-build/gulp-core-build/src/logging.ts]: Yes. Although we should make the gulp-util.File changes in a different PR

---
In reply to: [163129279](https://github.com/Microsoft/web-build-tools/pull/505#discussion_r163129279) [](ancestors = 163129279)

**Comment by nickpape-msft** [comment]
>     "gulp-util": "~3.0.7",

We still seem to use (at least the typings) in a few places. We should remove it as a separate PR. Goal of this PR was to get colors working properly in builds

---
In reply to: [359653500](https://github.com/Microsoft/web-build-tools/pull/505#issuecomment-359653500) [](ancestors = 359653500)

---
Refers to: core-build/gulp-core-build/package.json:43 in f4412b5. [](commit_id = f4412b5795766785055932963e10a86ed8bcde40, deletion_comment = False)

**Comment by nickpape-msft** [comment]
>     "node-forge": "~0.7.1",

See other reponses. That is out of scope for this change

---
In reply to: [359653565](https://github.com/Microsoft/web-build-tools/pull/505#issuecomment-359653565) [](ancestors = 359653565)

---
Refers to: core-build/gulp-core-build-serve/package.json:25 in f4412b5. [](commit_id = f4412b5795766785055932963e10a86ed8bcde40, deletion_comment = False)

**Pull request merged**